src/base/AnyMap.cpp

```diff
@@ -190,7 +190,11 @@ struct convert<Cantera::AnyValue> {
         if (node.IsScalar()) {
             // Scalar nodes are int, doubles, or strings
             std::string nodestr = node.as<std::string>();
-            if (isInt(nodestr)) {
+            if (node.Tag() == "!") {
+                // Undo conversion from string to other formats (done
+                // automatically by yaml-cpp, e.g. '12345' is converted to int)
+                target = nodestr;
+            } else if (isInt(nodestr)) {
```

Comment:
I believe this change will allow resolution of this comment: https://github.com/Cantera/cantera/blob/773523a4e314e4ce4bfec379b438c5ca55cebf5c/interfaces/cython/cantera/ctml2yaml.py#L1947-L1950 🎉

Reply:
Ok. Happy that it may help elsewhere!